ipld/read.go

```diff
@@ -50,17 +86,9 @@ func RetrieveData(
 	return extended, batchAdder.Commit()
 }
 
-type quadrant struct {
-	isLeftSubtree bool
-	from          int
-	to            int
-	rootCids      []cid.Cid
-}
-
-func pickRandomQuadrant(roots [][]byte) (*quadrant, error) {
+func pickQuadrant(qNumber int, roots [][]byte) (*quadrant, error) {
```

Comment:
1. Please do this refactoring in a different PR, as it's orthogonal.
2. As for the refactoring, the original `pickRandomQuadrant` is _good_ because it abstracts away specifics about the random range. Note that your refactoring doesn't actually check that the random number generated is between 1 and 4. It happens to work due to some implementation details (e.g. checking specifically odd then >2), but in the future those implementation details may change and cause spooky action at a distance.
3. If you want to be able to pick the specific quadrant, then you should have _both_ `pickQuadrant` and `pickRandomQuadrant`, the latter simply calling the former with a bounded random number.

Reply:
Ok, will revert changes with refactoring